report/4.Design/USB.tex

```diff
@@ -0,0 +1,26 @@
+\section{USB communication}
+\label{sec:usb}
+As described in the component diagram in section \ref{sec:architecture}, the system will consist of two seperate hardware platforms, that need to communicate.
+As the NXT is limited in communication to BlueTooth and USB, and bluetooth is too slow for a real-time system, the group has decided to use USB.
+According to the NXTOSEK documentation, the NXTOSEK library should also support built-in USB, through an easy to use API.
+When the group tested this functionality, it turned out that the driver for the PC/ RaspBerry no longer worked.
+
+However, since the driver is simply meant to be a translator between computers, and the devices attatched to them, the group decided to build their own driver, instead of just doing raw bit communication.
+
+The driver library will consist of three parts.
+The initialization part, which will perform a handshake with the device, telling the NXT that it should expect communication.
+This part posed a bit of a challenge, as it was not documented anywhere what the handshake consisted of, or that it even existed.
```

Comment:
Consider "This part posed a bit of a challenge, as it was not documented anywhere what the handshake consisted of, or that it even existed."

to 

"This part posed a bit of a challenge, as the handshake and its content was not documented, thus leaving us in the dark about its existence" el. lignede omformulering